src/packages/core/src/server.ts

```diff
@@ -59,28 +77,35 @@ export default class Server {
     return this.#status;
   }
 
-  constructor(serverOptions: ServerOptions = { flavor: DefaultFlavor }) {
-    const opts = (this.#options = serverOptionsConfig.normalize(serverOptions));
+  constructor(providerAndServerOptions: any = { flavor: DefaultFlavor }) {
+    this.#options = serverOptionsConfig.normalize(providerAndServerOptions);
+    this.#providerOptions = providerAndServerOptions;
+    this.#status = Status.ready;
+  }
+
+  private async initialize() {
     const connector = (this.#connector = ConnectorLoader.initialize(
-      serverOptions
+      this.#providerOptions
     ));
 
+    await connector.once("ready");
```

Comment:
It's possible we can move this to the bottom of the `initialize` function; I was try to prevent unnecessary dirty state if the initialize couldn't complete, but I can understand your desire here for performance rather than logical progression.